conanfile.py

```diff
@@ -459,13 +465,20 @@ def build(self):
                     args.append("\"%s_INCDIR=%s\"" % (var, self.deps_cpp_info[package].include_paths[-1]))
                 args += ["-D " + s for s in self.deps_cpp_info[package].defines]
 
+                def _remove_duplicate(l):
+                    seen = set()
+                    seen_add = seen.add
+                    for element in itertools.filterfalse(seen.__contains__, l):
+                        seen_add(element)
+                        yield element
+                
                 def _gather_libs(p):
                     libs = ["-l" + i for i in self.deps_cpp_info[p].libs]
                     libs += self.deps_cpp_info[p].sharedlinkflags
                     for dep in self.deps_cpp_info[p].public_deps:
                         libs += ["-L" + lpath for lpath in self.deps_cpp_info[dep].lib_paths]
                         libs += _gather_libs(dep)
-                    return libs
+                    return _remove_duplicate(libs)
```

Comment:
I'm not a python programmer, but it looks like `list(dict.fromkeys(libs))` is a more idiomatic way to remove duplicates from a list

Reply:
[dictionary before python 3.7 do not conserve insertion order.](https://docs.python.org/3.7/library/stdtypes.html#dict.values)